torch/utils/_cuda_trace.py

```diff
@@ -0,0 +1,52 @@
+import traceback
+import sys
+from typing import Callable, List, Tuple, Any
+
+
+class CallbackRegistry:
+    def __init__(self):
+        self.callback_list : List[Callable[[Tuple[Any, ...]], None]] = []
+
+    def add_callback(self, cb : Callable[[Tuple[Any, ...]], None]) -> None:
+        self.callback_list.append(cb)
+
+    def fire_callbacks(self, *args: int) -> None:
+        for cb in self.callback_list:
+            try:
+                cb(*args)
+            except Exception:
+                print(
+                    f"Callback registered with CUDA trace threw an exception:\n"
+                    f"{traceback.format_exc()}",
+                    file=sys.stderr
+                )
```

Comment:
This was my basic error reporting, I wonder if there's a better way of doing it? Now sure what PyTorch does in cases like these.

Also, it could be nice to include the name of the hook for which the callback failed.

Reply:
Done.